Core/Battler.gd

```diff
@@ -46,10 +46,19 @@ func _on_player_initialized() -> void:
 func _on_phase_changed(new_phase: Enums.Phase, _old_phase: Enums.Phase) -> void:
 	if new_phase == Enums.Phase.PLAYER_ATTACKING:
 		_on_player_start_turn()
+		
+	if new_phase == Enums.Phase.PLAYER_FINISHING:
+		_on_player_finished_turn()
 	
 	if new_phase == Enums.Phase.ENEMY_ATTACKING:
 		_on_enemy_start_turn()
 
+func _on_player_finished_turn():
+	if (!CardManager.is_discard_hand_signal_connected(_on_player_hand_discarded)):
+		CardManager.connect_discard_hand_signal(_on_player_hand_discarded)
```

Comment:
Maybe we can do this with the `on_card_container_initialized` signal instead of `on_phase_changed`. `CardPileUISetter` has a good example of this

Reply:
Ooo great idea. I will update.